projects/ids-enterprise-ng/src/lib/menu-button/soho-menu-button.component.ts

```diff
@@ -49,62 +60,180 @@ export class SohoMenuButtonComponent implements AfterViewInit, OnDestroy {
   /** The icon to be used. */
   @Input() icon: string;
 
+  @Input() set autoFocus(value: boolean) {
+    this.options.autoFocus = value;
+    if (this.menuButton) {
+      this.menuButton.settings.autoFocus = value;
+      this.markForRefresh();
```

Comment:
I don't believe you have to markForCheck for inputs. When an input is used this component should automatically be `markedForCheck`. The reason to call it manually is if you do something that needs to be reflected in the view but isn't going to automatically mark it.

Reply:
Although I don't think this hurts anything.